lib/factorial.rb

```diff
@@ -1,6 +1,12 @@
 # Computes factorial of the input number and returns it
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) linear, where n is the size of the number
+# Space complexity: O(1) no matter the size of n
 def factorial(number)
-  raise NotImplementedError
+  raise ArgumentError if !number
```

Comment:
What would happen if I put in `factorial("5")` and is that a reasonable case to support or guard against?